@somzzz please take a look at the autotester (e.g. https://auto-tester.puremagic.com/show-run.ghtml?projectid=1&runid=2431115&isPull=true), looks like a clerical issue:

```
Error - file 'ddmd/expression.d' contains trailing whitespace at line 15428
```

All: I'm also seeing a bazillion of these: ddmd/backend/ptrntab.c:5773: warning: dereferencing type-punned pointer will break strict-aliasing rules.

--- a/src/ddmd/expression.d
+++ b/src/ddmd/expression.d
@@ -15422,7 +15422,6 @@ extern (C++) final class CmpExp : BinExp
 
                 al = new CallExp(loc, al, arguments);
                 al = new CmpExp(op, loc, al, new IntegerExp(0));
-                al = al.semantic(sc);
 
                 arrayLowering = al;
             }
@@ -15524,7 +15523,11 @@ extern (C++) final class CmpExp : BinExp
         }
 
         //printf("CmpExp: %s, type = %s\n", e.toChars(), e.type.toChars());
-        if (arrayLowering) return arrayLowering;
+        if (arrayLowering)
+        {
+            arrayLowering = arrayLowering.semantic(sc);
+            return arrayLowering;
+        }
         return this;
     }
 

--- a/src/ddmd/expression.d
+++ b/src/ddmd/expression.d
@@ -15410,9 +15410,7 @@ extern (C++) final class CmpExp : BinExp
             }
             if ((t1.ty == Tarray || t1.ty == Tsarray) && (t2.ty == Tarray || t2.ty == Tsarray))
             {
-                semanticTypeInfo(sc, t1.nextOf());
-
-                // Lowering to template call
+                // Lower to object.__cmp(e1, e2)
                 Expression al = new IdentifierExp(loc, Id.empty);
                 al = new DotIdExp(loc, al, Id.object);
                 al = new DotIdExp(loc, al, Id.__cmp);
@@ -15423,9 +15421,7 @@ extern (C++) final class CmpExp : BinExp
                 arguments.push(e2);
 
                 al = new CallExp(loc, al, arguments);
-                al = al.semantic(sc);
-                
-                al = new CmpExp(op, loc, al, new IntegerExp(loc, 0, Type.tint32));
+                al = new CmpExp(op, loc, al, new IntegerExp(0));
                 al = al.semantic(sc);
 
                 arrayLowering = al;

